streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -356,20 +358,24 @@ public String queryableStoreName() {
     @Override
     public KTable<K, V> suppress(final Suppressed<K> suppressed) {
         final String name = builder.newProcessorName(SUPPRESS_NAME);
+        final String storeName = builder.newStoreName(SUPPRESS_NAME);
```

Comment:
yes. I have an interesting proposal to avoid this problem (forever), but I didn't want to introduce too much controversy right now.

Should we consider that as part of this PR? Alternatively, we can put in a minor patch for 2.1 independently.

Reply:
See https://issues.apache.org/jira/browse/KAFKA-7393

Maybe we should just add a name parameter to Suppressed. This should be an uncontroversial "merge" of Bill's any my KIPs.

Reply:
\cc @bbejeck @guozhangwang WDYT?